pkg/stream/consumer_base.go

```diff
@@ -194,16 +233,61 @@ func (c *baseConsumer) runConsumerCallback(ctx context.Context) error {
 	return nil
 }
 
+func (c *baseConsumer) shovelData(ctx context.Context) error {
```

Comment:
I'm not a big fan of using `shovel`, sounds a bit unprofessional to me :D

Reply:
renamed :D